packages/docs/docs/contributing/development-setup.md

```diff
@@ -0,0 +1,222 @@
+
+- **Yarn**: Version 4.9.1 or greater. Yarn is the package manager used by Actual.
+  - The project uses Yarn 4 workspaces (monorepo structure).
+
+- **Git**: Required for cloning the repository and version control.
+
+## Initial Setup
+
+1. Clone the Actual repository:
+
+   ```bash
+   git clone https://github.com/actualbudget/actual.git
+   cd actual
+   ```
+
+2. Install all dependencies:
+
+   ```bash
+   yarn install
+   ```
+
+   This will install dependencies for all packages in the monorepo.
+
+3. Verify your setup by running type checking:
+   ```bash
+   yarn typecheck
+   ```
+
+## Essential Development Commands
+
+All commands should be run from the **root directory** of the repository. Never run yarn commands from child workspace directories.
+
+### Type Checking
+
+```bash
+# Run TypeScript type checking (ALWAYS run before committing)
+yarn typecheck
+```
+
+### Linting and Formatting
+
+```bash
+# Check for linting and formatting issues
+yarn lint
+
+# Auto-fix linting and formatting issues
+yarn lint:fix
+```
+
+### Testing
+
+```bash
+# Run all tests across all packages
+yarn test
+
+# Run tests without cache (for debugging)
+yarn test:debug
+```
+
+For more details on testing, see the [Testing Guide](./testing.md).
+
+### Starting Development Servers
+
+```bash
+# Start browser development server
+yarn start
+# or explicitly:
+yarn start:browser
+
+# Start with sync server (for testing sync functionality)
+yarn start:server-dev
+
+# Start desktop app development
+yarn start:desktop
+```
+
+### Building
+
+```bash
+# Build browser version
+yarn build:browser
+
+# Build desktop app
+yarn build:desktop
+
+# Build API package
+yarn build:api
+
+# Build sync server
+yarn build:server
+```
+
+## Workspace Structure
+
+Actual uses Yarn workspaces to manage a monorepo with multiple packages. The main packages are:
+
+- `loot-core` - Core application logic
+- `desktop-client` (aliased as `@actual-app/web`) - React-based UI
+- `desktop-electron` - Electron wrapper for desktop app
+- `api` (aliased as `@actual-app/api`) - Public API package
+- `sync-server` (aliased as `@actual-app/sync-server`) - Synchronization server
+- `component-library` (aliased as `@actual-app/components`) - Reusable UI components
+- `crdt` (aliased as `@actual-app/crdt`) - CRDT implementation for sync
+- `plugins-service` - Service for handling plugins
+- `eslint-plugin-actual` - Custom ESLint rules
+- `docs` - Documentation website
+
+For detailed information about each package, see the [Project Structure](./project-details/index.md) documentation.
+
+## Running Workspace-Specific Commands
+
+To run commands for a specific workspace, use:
+
+```bash
+yarn workspace <workspace-name> run <command>
+```
+
+Examples:
+
+```bash
+# Run tests for loot-core
+yarn workspace loot-core run test
+
+# Start the docs development server
+yarn workspace docs start
+
+# Build the API package
+yarn workspace @actual-app/api build
+```
```

Comment:
This is already documented in project-details, I don't think we need it in both places. I prefer it in this file.